src/gd_kotlin_configuration.cpp

```diff
@@ -97,11 +121,235 @@ void GdKotlinConfiguration::set_max_string_size(int p_max_string_size) {
+                ":" +
+                jvm_debug_port};
+        args->option(debug_command.utf8());
+    }
+
+    if (!jvm_jmx_port.empty()) {
+        String port_command{"-Dcom.sun.management.jmxremote.port=" + jvm_jmx_port};
+        String rmi_port{"-Dcom.sun.management.jmxremote.rmi.port=" + jvm_jmx_port};
+        args->option("-Djava.rmi.server.hostname=127.0.0.1");
+        args->option("-Dcom.sun.management.jmxremote");
+        args->option(port_command.utf8());
+        args->option(rmi_port.utf8());
+        args->option("-Dcom.sun.management.jmxremote.local.only=false");
+        args->option("-Dcom.sun.management.jmxremote.authenticate=false");
+        args->option("-Dcom.sun.management.jmxremote.ssl=false");
+#ifdef DEBUG_ENABLED
+        LOG_VERBOSE(vformat("Started JMX on port: %s", jvm_jmx_port))
+#endif
+    }
+}
+
+jni::Jvm::Type GdKotlinConfiguration::vm_type_from_string(const String& vm_type_as_string) {
+    if (vm_type_as_string == jvm_string_identifier) {
+        return jni::Jvm::JVM;
+    } else if (vm_type_as_string == graal_native_image_string_identifier) {
+        return jni::Jvm::GRAAL_NATIVE_IMAGE;
+    } else if (vm_type_as_string == art_string_identifier) {
+        return jni::Jvm::ART;
+    } else {
+        LOG_WARNING("Wrong JVM type in config, fallback to classic JVM !")
+        return jni::Jvm::JVM;
+    }
+}
+
+GdKotlinConfiguration GdKotlinConfiguration::load_from_json() {
+    GdKotlinConfiguration configuration;
+    if (FileAccess::exists(gd_kotlin_configuration_path)) {
+        FileAccessRef configuration_access_read{FileAccess::open(gd_kotlin_configuration_path, FileAccess::READ)};
+        configuration = GdKotlinConfiguration::from_json(configuration_access_read->get_as_utf8_string());
+        configuration_access_read->close();
+    } else {
+#ifdef TOOLS_ENABLED
+        FileAccessRef file = FileAccess::open(gd_kotlin_configuration_path, FileAccess::WRITE);
+        configuration = GdKotlinConfiguration();
+        file->store_string(configuration.to_json());
+        file->close();
+#else
+        LOG_ERROR(vformat("Cannot find Godot Kotlin configuration file at: %s. Falling back to default configuration.", gd_kotlin_configuration_path))
+        configuration = GdKotlinConfiguration();
+#endif
+    }
+#ifdef __ANDROID__
+    configuration.set_vm_type(jni::Jvm::ART);
+#endif
+    return configuration;
+}
+
+void GdKotlinConfiguration::override_json_config_with_cmd_args(GdKotlinConfiguration* r_configuration) {
+    const List<String>& cmdline_args{OS::get_singleton()->get_cmdline_args()};
+    const String& arg_prefix{"--"};
+
+    for (int i = 0; i < cmdline_args.size(); ++i) {
+        const String cmd_arg{cmdline_args[i]};
+        if (cmd_arg.find(arg_prefix + vm_type_identifier) >= 0) {
+            String vm_type_as_string;
+            if (split_jvm_debug_argument(cmd_arg, vm_type_as_string) == OK) {
+#ifdef __ANDROID__
```

Comment:
Just for practical reasons for when we will also support IOS.
I would add a `sanitize_config` function where we put all the OS-related preprocessors so they are all in one place and call it inside `load_gd_kotlin_configuration_from_json_and_args`

Reply:
I suggest doing this when doing #258 and we refactor the `GDKotlin` class because i guess we'll rework this part with that as well.